src/Initializer/InitProcedure/InitModel.cpp

```diff
@@ -25,66 +26,6 @@ using namespace seissol::initializer;
 using Material_t = seissol::model::Material_t;
 using Plasticity = seissol::model::Plasticity;
 
-/*
-Assigns the given value to the target object, initializing the memory in the process.
-
-NOTE: std::copy (or the likes) do not work here, since they do not initialize the _vptr for virtual
-function calls (rather, they leave it undefined), since they do merely assign `value` to `target`.
-*/
-
-template <typename T>
-static void initAssign(T& target, const T& value) {
-  if constexpr (std::is_trivially_copyable_v<T>) {
-    // if the object is trivially copyable, we may just memcpy it (it's safe to do that in this
-    // case).
-    std::memcpy(&target, &value, sizeof(T));
-  } else {
-    // otherwise, call the class/struct initializer.
-    // problem: we may have an array here... So we unwrap it.
-    if constexpr (std::is_array_v<T>) {
-      // unwrap array, dimension by dimension...
-      // example: T[N][M] yields SubT=T[M]
-      using SubT = std::remove_extent_t<T>;
-      auto subExtent = std::extent_v<T>;
-
-      // for now, init element-wise... (TODO(David): we could look for something faster here, in
-      // case it should ever matter)
-      for (size_t i = 0; i < subExtent; ++i) {
-        initAssign<SubT>(target[i], value[i]);
-      }
-    } else {
-      // now call new here.
-      new (&target) T(value);
-    }
-  }
-  // (these two methods cannot be combined, unless we have some way for C-style arrays, i.e. S[N]
-  // for <typename S, size_t N>, to use a copy constructor as well)
-}
-
-template <typename T>
-static void synchronize(const seissol::initializers::Variable<T>& handle) {
```

Comment:
so this function was never called? Or did you forget to remove the previous call?

Reply:
It was, in the same file.

I kept the calls from FORTRAN, but changed the iteration method—before it iterated over all mesh cells, but now it iterates over the whole LTS tree (thus including duplicates). Hence, we may probably remove it.

Reply:
I've removed the calls in InitModel; the duplicates were really initialized before that already, because we went through the whole LTS tree there.